pkgs/by-name/ch/chatd/package.nix

```diff
@@ -0,0 +1,68 @@
+{ buildNpmPackage
+, lib
+, electron
+, fetchFromGitHub
+, ollama
+, makeWrapper
+, nix-update-script
+}:
+
+buildNpmPackage rec {
+  pname = "chatd";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "BruceMacD";
+    repo = "chatd";
+    rev = "v${version}";
+    hash = "sha256-3VtF2vLs+bdT9VX3A+A8hntzNZZ+SL5UZqs5Cm/1264=";
+  };
+
+  npmDepsHash = "sha256-RwugV6548zaxuAHoWImNKKokPkMnCWQWaUvQhH/AxGU=";
+
+  dontNpmBuild = true;
+
+  npmFlags = [ "--ignore-scripts" ];
+
+  nativeBuildInputs = [
+    makeWrapper
+    electron
+  ];
+
+  postPatch = ''
+    # basically https://github.com/BruceMacD/chatd/pull/35
+    # requires a new release to work with remote ollama instances
+    substituteInPlace src/service/ollama/ollama.js \
+      --replace 'this.host = "http://127.0.0.1:11434";' 'this.host = process.env.OLLAMA_HOST || "http://127.0.0.1:11434";'
+  '';
+
+  buildPhase = ''
+  runHook preBuild
+
+  runHook postBuild
+  '';
+
+  installPhase = ''
```

Comment:
If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md